plugins/importexport/crossref/CrossrefInfoSender.inc.php

```diff
@@ -189,12 +121,77 @@ function _getJournals() {
 			if ($doiPrefix) {
 				$journals[] = $journal;
 			} else {
-				$this->addExecutionLogEntry(__('plugins.importexport.crossref.senderTask.warning.noDOIprefix', array('path' => $journal->getPath())), SCHEDULED_TASK_MESSAGE_TYPE_WARNING);
+				$this->addExecutionLogEntry(__('plugins.importexport.common.senderTask.warning.noDOIprefix', array('path' => $journal->getPath())), SCHEDULED_TASK_MESSAGE_TYPE_WARNING);
 			}
 			unset($journal);
 		}
 		return $journals;
 	}
 
+	/**
+	 * Update the status and construct an array of the objects to be deposited
+	 * @param $unregisteredObjects array Array of all not fully registered objects
+	 * @param $journal Journal
+	 * @param $notify boolean
+	 * @return array Array of objects to be deposited
+	 */
+	function _getObjectsToBeDeposited($unregisteredObjects, $journal, &$notify) {
+		$plugin = $this->_plugin;
+		$objectsToBeDeposited = array();
+		foreach ($unregisteredObjects as $object) {
+			$plugin->updateDepositStatus($journal, $object);
+			// get the current object status
+			$currentStatus = $object->getData($plugin->getDepositStatusSettingName());
+			// deposit only not submitted objects
+			if (!$currentStatus) {
+				array_push($objectsToBeDeposited, $object);
+			}
+			// check if the new status after the update == failed to notify the users
+			$newStatus = $object->getData($plugin->getDepositStatusSettingName());
+			if (!$notify && $newStatus == CROSSREF_STATUS_FAILED && $currentStatus != CROSSREF_STATUS_FAILED) {
+				$notify = true;
+			}
+		}
+		return $objectsToBeDeposited;
+	}
+
+	/**
+	 * Register objects
+	 * @param $objects array
+	 * @param $filter string
+	 * @param $journal Journal
+	 */
+	function _registerObjects($objects, $filter, $journal) {
+		$plugin = $this->_plugin;
+		// export XML
+		$exportXml = $plugin->exportXML($objects, $filter, $journal);
+		// Write the XML to a file.
+		$exportFileName = $plugin->getExportPath() . date('Ymd-His') . '.xml';
```

Comment:
Wow, good point! I wonder it worked without problems till now! Or maybe it didn't and it happened already that a journal registered the DOIs from another journal in the same installation, in case the both exports were done at the same time. But maybe this was never the case. Phew... :-)